web-frontend/src/pages/SelectAccount/SelectAccount.tsx

```diff
@@ -3,59 +3,54 @@ import { FiUser, FiPlus } from 'react-icons/fi';
 import { Button } from '@massalabs/react-ui-kit/src/components/Button/Button';
 import { AccountSelectorButton } from '@massalabs/react-ui-kit/src/components/AccountSelector/AccountSelector';
 import { MassaToken } from '@massalabs/react-ui-kit/src/components/Icons/Svg/SvgComponent/MassaToken';
-import { useQuery } from '@tanstack/react-query';
-import { getAllAccounts } from '../../api/account';
-import { accountType } from '../../api/types';
 import { toMAS } from '@massalabs/massa-web3';
+import { Link, useNavigate } from 'react-router-dom';
+import { goToErrorPage, routeFor } from '../../utils';
+import useResource from '../../custom/api/useResource';
+import { AccountObject } from '../../models/AccountModel';
 
 export default function SelectAccount() {
-  const accounts = useQuery({
-    queryKey: ['accounts'],
-    queryFn: getAllAccounts,
-  });
+  // If no account, redirect to welcome page
+  const navigate = useNavigate();
 
-  const accountBaseProps = {
-    avatar: <FiUser className="text-neutral h-6 w-6" />,
-    icon: <MassaToken size={24} />,
-  };
+  const { error, data = [] } = useResource<AccountObject[]>('accounts');
 
-  const buttonProps = {
-    onClick: () => {
-      console.log('clicked');
-    },
-    preIcon: <FiPlus />,
-  };
+  if (error) goToErrorPage(navigate);
+
+  if (!data.length) {
+    navigate(routeFor('welcome'));
+  }
 
   const defaultFlex = 'flex flex-col justify-center items-center align-center';
   return (
     <LandingPage>
-      <div
-        className={`${defaultFlex} 
-                      h-screen`}
-      >
+      <div className={`${defaultFlex} h-screen`}>
         <div className="w-1/2">
           <div className="mb-6">
-            <p className="mas-banner text-neutral"> Hey !</p>
+            <p className="mas-banner text-neutral"> Hey!</p>
           </div>
           <div className="mb-6">
             <label className="mas-body text-neutral" htmlFor="account-select">
               Select an account
             </label>
           </div>
           <div id="account-select" className="w-full flex flex-col">
-            {accounts.data?.map((account: accountType) => (
-              <div className="mb-4">
+            {data.map((account: AccountObject) => (
```

Comment:
i think you should keep the "?" after data, it may be undefined in case of http error

Reply:
normally it's handled with data = [] above